### ISSUE TITLE
Polish translation

### ISSUE THREAD

**Issue by jsbien** [Codepoints/Codepoints.net#74]
I am interested in preparing the Polish  translation, hopefully with the help of my colleagues and students.

Please create a new translation at https://crowdin.com/project/codepoints.

Best regards

Janusz

**Comment by Boldewyn** [comment]
Hello, and thank you for helping out with translating!

I've set up a Polish translation at Crowdin:

https://crowdin.com/project/codepoints/pl

Before you start, please note, that translations must be provided under the terms of the Creative Commons Attribution license as laid out on [the "about" page](https://codepoints.net/about#this_site). If you're fine with that, I'll happily merge it (and amend said page to point out your contribution).

You can localize _almost_ everything with the two .pot files over at Crowdin. However, there is another source file, that'll  need your attention for a complete translation. It's a PHP file, that contains translated names for Unicode's special phrases. The German one is here, for reference:

https://github.com/Codepoints/Codepoints.net/blob/master/codepoints.net/locale/de/unicodeinfo.php

The Polish localization would live in `/codepoints.net/locale/pl/unicodeinfo.php`. I suggest, we start localization without this file and get to it again, when we're close to finishing the .pot files.

From time to time there will be changes and/or updates to the source files. How can I contact you in this case? Do you prefer e-mail, Crowdin discussions, or something else?

If you have any questions, encounter any problems or have any kind of feedback, just get back at me any time here on this page or via e-mail, boldewyn@gmail.com.

**Comment by jsbien** [comment]
Quote/Cytat - Manuel Strehl <notifications@github.com> (Mon 15 Feb  
2016 10:23:19 PM CET):

> Hello, and thank you for helping out with translating!
>
> I've set up a Polish translation at Crowdin:
>
> https://crowdin.com/project/codepoints/pl

Thanks.

>
> Before you start, please note, that translations must be provided  
> under the terms of the Creative Commons Attribution license as laid  
> out on [the "about" page](https://codepoints.net/about#this_site).  
> If you're fine with that, I'll happily merge it (and amend said page  
> to point out your contribution).

CA-By is OK for me.

>
> You can localize _almost_ everything with the two .pot files over at  
> Crowdin. However, there is another source file, that'll  need your  
> attention for a complete translation. It's a PHP file, that contains  
> translated names for Unicode's special phrases. The German one is  
> here, for reference:
>
> https://github.com/Codepoints/Codepoints.net/blob/master/codepoints.net/locale/de/unicodeinfo.php
>
> The Polish localization would live in  
> `/codepoints.net/locale/pl/unicodeinfo.php`. I suggest, we start  
> localization without this file and get to it again, when we're close  
> to finishing the .pot files.

OK

>
> From time to time there will be changes and/or updates to the source  
> files. How can I contact you in this case? Do you prefer e-mail,  
> Crowdin discussions, or something else?

I prefer mail.

>
> If you have any questions, encounter any problems or have any kind  
> of feedback, just get back at me any time here on this page or via  
> e-mail, boldewyn@gmail.com.

OK

Best regards

Janusz

-- 
Prof. dr hab. Janusz S. Bień -  Uniwersytet Warszawski (Katedra  
Lingwistyki Formalnej)
Prof. Janusz S. Bień - University of Warsaw (Formal Linguistics Department)
jsbien@uw.edu.pl, jsbien@mimuw.edu.pl, http://fleksem.klf.uw.edu.pl/~jsbien/

**Comment by jsbien** [comment]
Is there a way to access context from crowin.com? I miss it very much. 

I had some localisation experience some time ago, but don't remember now details and I'm confused by things like 'a %svery good introduction%s to'. What the 2 variables %s stand for? Or it is a single variable in some notation?

Regards

Janusz

**Comment by Boldewyn** [comment]
As far as I see, Crowdin shows the context path from the .po file under each string, but it doesn't go the extra mile to add the actual source snippet. However, the context paths can be translated directly to URLs on Github, e.g.

    codepoints.net/views/footer.php
    ->
    https://github.com/Codepoints/codepoints.net/tree/master/codepoints.net/views/footer.php
                          ----------------------------------^

That is, prefix with `https://github.com/Codepoints/codepoints.net/tree/master/`.

----

When you encounter two `%s` in close succession, it's mostly HTML tags, that will get replaced. So, a string

     %sbold text%s

will, e.g., read like this on the page:

    <b>bold text</b>

There might be some replacement strings, that are a bit more arcane. They are documented on [PHP's `sprintf()`](http://php.net/sprintf) page. One useful is `%04X`, that formats a number as uppercase hex digit, optionally padded with 0's. That's the `U+` notation for codepoints: `U+0001 = U+%04X`

PHP has also a special feature: If you need to re-order replacements, you can do so with the special notation `%1$s` for the first replacement, `%2$s` for the second and so on.

----

On https://beta.codepoints.net I have a staging/preview area. When you want to test your localization, just drop me a line. I'll fetch the translations from Crowdin then and put them live there. (At the moment I'll do it manually, so you'll have to bear with me, until it's updated.)

**Comment by jsbien** [comment]
I've made a draft translation of both po files. I think the translation is complete, but crowdin reports only 89%. Anyway I think it's time for  `/codepoints.net/locale/pl/unicodeinfo.php`.

Regards

Janusz

**Comment by Boldewyn** [comment]
Fantastic! I updated https://beta.codepoints.net/?lang=pl, where you can see the result.

About those 89%: In my view of `messages.pot` at Crowdin, in the left column, there are the strings listed with untranslated first. There is exactly one page (the first) with untranslated strings. Perhaps Crowdin changed the sorting order, while you prepared the translation, or it got the pagination mixed up?

I'll put together some info about how we could handle `unicodeinfo.php` tomorrow.

While I prepared the code for the Polish translation, I remembered there is also a second location in the source files, where localized text appears: in the folder [`codepoints.net/data`](https://github.com/Codepoints/Codepoints.net/tree/master/codepoints.net/data/). There I put extra files for _some_ characters, where I added additional information (e.g., why there is a `MAN IN BUSINESS SUIT LEVITATING` character). Those files are all small Markdown files (i.e., the same syntax as the one in this bu report).

I placed a "How to contribute" guide in that folder:

https://github.com/Codepoints/Codepoints.net/blob/master/codepoints.net/data/00_HOW_TO_CONTRIBUTE.md

We could try translating these files, if you like, as laid out in that document. But since this is pioneering work, the process might not be suited for bulk translation. If it's too cumbersome this way, I'll try to come up with another solution.

**Comment by jsbien** [comment]
Quote/Cytat - Manuel Strehl <notifications@github.com> (Tue 16 Feb  
2016 11:37:43 PM CET):

> Fantastic! I updated https://beta.codepoints.net/?lang=pl, where you  
> can see the result.

Thanks! I see already some mistakes, including simple typos.

>
> About those 89%: In my view of `messages.pot` at Crowdin, in the  
> left column, there are the strings listed with untranslated first.  
> There is exactly one page (the first) with untranslated strings.  
> Perhaps Crowdin changed the sorting order, while you prepared the  
> translation, or it got the pagination mixed up?

I see them now too, somehow I've missed them later.

> I'll put together some info about how we could handle  
> `unicodeinfo.php` tomorrow.

OK, there is no hurry. Tomorrow a new semester starts and I have a lot  
of teaching duties, so from now on I will work slooow :-(
>
> While I prepared the code for the Polish translation, I remembered  
> there is also a second location in the source files, where localized  
> text appears: in the folder  
> [`codepoints.net/data`](https://github.com/Codepoints/Codepoints.net/tree/master/codepoints.net/data/). There I put extra files for _some_ characters, where I added additional information (e.g., why there is a `MAN IN BUSINESS SUIT LEVITATING` character). Those files are all small Markdown files (i.e., the same syntax as the one in this bu  
> report).
>
> I placed a "How to contribute" guide in that folder:
>
> https://github.com/Codepoints/Codepoints.net/blob/master/codepoints.net/data/00_HOW_TO_CONTRIBUTE.md
>
> We could try translating these files, if you like, as laid out in  
> that document. But since this is pioneering work, the process might  
> not be suited for bulk translation. If it's too cumbersome this way,  
> I'll try to come up with another solution.

There also some general localisation problems which require attention.  
Polish words and sentences are usually longer and an example is on the  
start page: "Wyszukiwanie zaawansowane" doesn't fit the available  
space. For the time being I have no idea for a shorter wording.

Regards

Janusz

-- 
Prof. dr hab. Janusz S. Bień -  Uniwersytet Warszawski (Katedra  
Lingwistyki Formalnej)
Prof. Janusz S. Bień - University of Warsaw (Formal Linguistics Department)
jsbien@uw.edu.pl, jsbien@mimuw.edu.pl, http://fleksem.klf.uw.edu.pl/~jsbien/

**Comment by Boldewyn** [comment]
I've merged your update and put it on beta.codepoints.net, too.

I also updated the code and data to show Wikipedia abstracts of codepoints in the right language. (Until today it was always the English version.) You should see that now on codepoint pages. (If the non-English Wikipedias have no entry for a codepoint, the English abstract is shown as fallback.)

----

The problem with word lengths is well-known to me. German translators face it, too, compared to English text. :smile: But as far as I can see, it's only the home page, that has a problem with overflowing text. I think, I'll try reducing the font size there a bit. Have you spotted other pages with problems?

----

Translating `unicodeinfo.php`: I've put a plain, untranslated file on Github:

https://github.com/Codepoints/Codepoints.net/blob/pl/codepoints.net/locale/pl/unicodeinfo.php

You could edit it right here (pen icon in the top right) and then save/commit. The file is basically a long map of Unicode properties onto their nice names for display. The values after the `=>` arrows are meant to be translated, they will be displayed on page.

It's presumably not necessary to translate the whole file from top to bottom. There are some quick wins (like "yes"/"no" fields), and apart from that the most interesting properties for translation are `categories` (properties of a codepoint), `gc` (general category), `sc` (script) and `blk` (Unicode block). I have translated only some of the other fields to German myself. If, however, you want to translate the whole thing, it's all the better :smile:.

----

Time constraints: I am eager to put the translation live as soon as possible. But of course, doing this in your spare time, I don't intend to put any schedule on this. I'd say, we move forward at your pace, I keep the beta site in sync, and I publish it as soon, as we both feel comfortable with the result.

So, again thank you a lot for the time you put into this! (And hopefully a productive new semester with eager students!)

**Comment by jsbien** [comment]
On Wed, Feb 17 2016 at 22:34 CET, notifications@github.com writes:
> I've merged your update and put it on beta.codepoints.net, too.

Thanks! The next Wednesday will be a good day for the next merge.

>
> I also updated the code and data to show Wikipedia abstracts of
> codepoints in the right language. (Until today it was always the
> English version.) You should see that now on codepoint pages. (If the
> non-English Wikipedias have no entry for a codepoint, the English
> abstract is shown as fallback.)

Thanks, it works, cf. https://beta.codepoints.net/U+0142.

BTW, the search  for "l with stroke" doesn't work - I created an issue.

>
> ----------------------------------------------------------------------
>
> The problem with word lengths is well-known to me. German translators
> face it, too, compared to English text. 
> :smile:
> But as far as I can see, it's only the home page, that has a problem
> with overflowing text. I think, I'll try reducing the font size there
> a bit. Have you spotted other pages with problems?

I haven't yet looked at other pages. If I notice a problem, I will
report it on the crowdin discussion.

>
> ----------------------------------------------------------------------
>
> Translating unicodeinfo.php: I've put a plain, untranslated file on
> Github:
>
> https://github.com/Codepoints/Codepoints.net/blob/pl/codepoints.net/locale/pl/unicodeinfo.php

The link works, but when I make 'git pull' I don't get the blob
directory. Is it intentional? I'm a regular user of mercurial with
almost zero knowledge of git.

[...]

>
> Time constraints: I am eager to put the translation live as soon as
> possible. But of course, doing this in your spare time, I don't intend
> to put any schedule on this. I'd say, we move forward at your pace, I
> keep the beta site in sync, and I publish it as soon, as we both feel
> comfortable with the result.

I will concentrate first on correcting my mistakes visible on the start
page and improving some wordings. As I said earlier, I intend to finish
this task by Wednesday. On Thursdays I'm teaching Unicode and I would
like to discuss the changes with my students.

Last but not least, is there a good tutorial for crowdin? I'm quite
often rather confused. What is the best place to ask question?

Best regards

Janusz

-- 
                           ,   
Prof. dr hab. Janusz S. Bien -  Uniwersytet Warszawski (Katedra Lingwistyki Formalnej)
Prof. Janusz S. Bien - University of Warsaw (Formal Linguistics Department)
jsbien@uw.edu.pl, jsbien@mimuw.edu.pl, http://fleksem.klf.uw.edu.pl/~jsbien/

**Comment by Boldewyn** [comment]
Getting the `unicodeinfo.php` file via Git: "blob" is Git terminology for a file-like structure. Github uses that in its URLs, when refering to a file. (You'll notice a "/tree/" in the URL, when Github shows a folder's content. That's the second of three special types in Git.)

To put it in a nutshell: Getting to the file in a shell:

```sh
# clone the repository:
$ git clone git@github.com:Codepoints/Codepoints.net.git && cd Codepoints.net
# check out the branch "pl" (assuming a newer Git version):
$ git checkout pl
Branch pl set up to track remote branch pl from origin.
Switched to a new branch 'pl'
# that message above tells you, that Git created the local branch "pl"
# automatically from the branch "pl" on Github
# Now you have the file at ./codepoints.net/locale/pl/unicodeinfo.php
```

----

> I will concentrate first on correcting my mistakes visible on the start page and improving some wordings. As I said earlier, I intend to finish this task by Wednesday. On Thursdays I'm teaching Unicode and I would like to discuss the changes with my students.

Perfect!

> Last but not least, is there a good tutorial for crowdin? I'm quite often rather confused. What is the best place to ask question?

Frankly, I don't know yet. But I'll take a look around. If it helps, I found out by chance yesterday, that the program [Poedit](http://poedit.net/) has settings to connect automatically to Crowdin. I haven't explored that in depth, but if you rather work locally, that might be something to look into in more depth.

**Comment by jsbien** [comment]
On Thu, Feb 18 2016 at 23:46 CET, notifications@github.com writes:

[...]

> # check out the branch "pl" (assuming a newer Git version):
> $ git checkout pl
> Branch pl set up to track remote branch pl from origin.
> Switched to a new branch 'pl'
> # that message above tells you, that Git created the local branch "pl"
> # automatically from the branch "pl" on Github
> # Now you have the file at ./codepoints.net/locale/pl/unicodeinfo.php
> ----------------------------------------------------------------------

Thanks. After selecting the correct branch I have the file.


[...]

> Frankly, I don't know yet. But I'll take a look around. If it helps, I
> found out by chance yesterday, that the program Poedit has settings to
> connect automatically to Crowdin. I haven't explored that in depth,
> but if you rather work locally, that might be something to look into
> in more depth.

This is a good news as I happened to use already poedit for some tasks
earlier. However in poedit 1.8.7 from Debian testing I don't see the
option to connect to Crowdin :-)

**Comment by jsbien** [comment]
I don't see in the po files strings for the form questions, e.g. "Where does the character appear usually?". Am I missing something obvious?

**Comment by Boldewyn** [comment]
> However in poedit 1.8.7 from Debian testing I don't see the option to connect to Crowdin

My Poedit 1.8.4 has the option under Edit > Preferences in the tab "Accounts". But as I said, I haven't tested it any further than just logging in.

> I don't see in the po files strings for the form questions, e.g. "Where does the character appear usually?". Am I missing something obvious?

Ah, good catch! It seems, that the Perl parser of `xgettext` misses some string instances. (`xgettext` has no Javascript parser, so the suggestion is to tell it to assume Perl.) There was an `xgettext` clone for Javascript written in Javascript. I'll try finding it and updating `js.pot` with it.

**Comment by jsbien** [comment]
Poedit for Windows has the Crowdin integration, Debian version is missing it for some reasons. I will do some experiments with the Windows version to find whether it is convenient.

**Comment by Boldewyn** [comment]
I updated beta.codepoints.net with the latest translations from Crowdin. As far as I see, the status is good enough to start. We can refine later, and add the missing parts. Fallback will be English, so there are no template tags or such leaking.

I included an attribution on the "about" page: https://beta.codepoints.net/about#attribution, "Attribution & Credits", second paragraph. Is this OK? If you want to provide a link, I'd add that, too.

What do you mean?

**Comment by jsbien** [comment]
Quote/Cytat - Manuel Strehl <notifications@github.com> (nie, 20 mar  
2016, 22:30:23):

> I updated beta.codepoints.net with the latest translations from  
> Crowdin. As far as I see, the status is good enough to start. We can  
> refine later, and add the missing parts. Fallback will be English,  
> so there are no template tags or such leaking.

OK

>
> I included an attribution on the "about" page:  
> https://beta.codepoints.net/about#attribution, "Attribution &  
> Credits", second paragraph. Is this OK? If you want to provide a  
> link, I'd add that, too.

Please add the link to  
http://bc.klf.uw.edu.pl/view/creators/Bie==0144=3AJanusz_S=2E=3A=3A.html.

Unfortunately my personal page is no longer maintained.

Best regards

Janusz

-- 
Prof. dr hab. Janusz S. Bień -  Uniwersytet Warszawski (Katedra  
Lingwistyki Formalnej)
Prof. Janusz S. Bień - University of Warsaw (Formal Linguistics Department)
jsbien@uw.edu.pl, jsbien@mimuw.edu.pl, http://fleksem.klf.uw.edu.pl/~jsbien/

**Comment by jsbien** [comment]
I have changed my mind about the link. I propose now to extend
`The Polish translation is kindly provided by professor Janusz S. Bień`
with something like
`He uses the terminology introduced in his paper "Standard Unicode 4.0. Wybrane pojęcia i terminy" (http://bc.klf.uw.edu.pl/114/) and the subsequent publications (http://bc.klf.uw.edu.pl/view/creators/Bie==0144=3AJanusz_S=2E=3A=3A.html)'

**Comment by Boldewyn** [comment]
Perfect! I'll add it as soon as I have a text editor nearby.

**Comment by jsbien** [comment]
The attribution: please skip the professor title.

The translation: please upload the current translation to beta.

**Comment by Boldewyn** [comment]
Will do as soon as possible. Please excuse, that I'm a little behind in communications as of late! I'm rather busy at the moment, but that will improve towards end of April.

**Comment by jsbien** [comment]
Quote/Cytat - Manuel Strehl <notifications@github.com> (Thu 07 Apr  
2016 02:06:57 PM CEST):

> Will do as soon as possible. Please excuse, that I'm a little behind  
> in communications as of late! I'm rather busy at the moment, but  
> that will improve towards end of April.

OK, there is no hurry.

**Comment by Boldewyn** [comment]
Done (finally!). Sorry for the long delay!

On the bright side: I added some targets to the Makefile to automatically update translation files and pull the translations from Crowdin. So, I should be quicker with update requests from now on :smile:

**Comment by jsbien** [comment]
Unfortunately the corrections suggested by Michał and made by me in March has vanished somewhere on the way. I'm stll confused very much about the workflow. Do you keep data for the beta version in a separate repository?

I understand there is no longer a pl branch. What I should do to pull the current version? Now I get this:

> Your configuration specifies to merge with the ref 'refs/heads/pl'
from the remote, but no such ref was fetched.

BTW, I think it would be more convenient to keep this issue open.

**Comment by Boldewyn** [comment]
> Unfortunately the corrections suggested by Michał and made by me in March has vanished somewhere on the way.

Uh-oh! That is unfortunate! Hard to tell, what went wrong. I downloaded the .po files from Crowdin after hitting the "build project" button in the admin interface. As I understand the Crowdin docs, that should be the standard procedure...

The commit for the current state of translation: https://github.com/Codepoints/Codepoints.net/commit/dec79f29d10b52753fbe054f2f6d4255e50b3c03

There are many deletions and additions, that are only style changes from `msgstr ""` to `msgstr "\n"` by Crowdin.

By the way: beta.codepoints.net is in an older state, I published the changes directly to codepoints.net.

> I understand there is no longer a pl branch. What I should do to pull the current version?

    git checkout master # needed only once
    git pull

I merged the translation back into the `master` branch. Beta: This is on demand any commit that needs testing. Can be in a feature branch or a commit in `master`.

> BTW, I think it would be more convenient to keep this issue open.

OK.

**Comment by jsbien** [comment]
I've pulled the current version ans started the development server. However switching languages doesn't work, always the English version is displayed. Bug or feature?

**Comment by Boldewyn** [comment]
That would be a terrible feature indeed :(. I assume it has something to do with cookies or the PHP gettext library. I'll investigate that and write back, when I've found something.

**Comment by jsbien** [comment]
OK. I can try also to set up the Apache server.

Another matter: which Makefile target I should use to check that the right version of my translation is downloaded?

**Comment by jsbien** [comment]
At https://crowdin.com/translate/codepoints/7101/en-pl#q=Teoretycznie I see my corrections (including the ones made with poedit several minutes ago) as suggestions, the translation itself is not corrected. What am I doing wrong?

**Comment by Boldewyn** [comment]
> which Makefile target I should use to check that the right version of my translation is downloaded?

Unfortunately, none. The Makefile targets `l10n-pull` and `l10n-push` need a Crowdin API key. But you could download the files

    https://crowdin.com/project/codepoints/pl/7102/download > pl/LC_MESSAGES/js.po
    https://crowdin.com/project/codepoints/pl/7101/download > pl/LC_MESSAGES/messages.po

and then do a

    git status

to see, if they changed. There seems to be a [command-line tool](https://crowdin.com/page/cli-tool) to ease that a bit, but I haven't tried it yet.

>  as suggestions, the translation itself is not corrected.

Behind the translations are little check marks. One is green, that's the approved one. (If you cannot see them, I messed something up with making you a proofreader. Then tell me please!)

It should work to click the hook for the correct translation to set this one as the correct one.
